ldap-authenticator/src/main/java/org/xwiki/contrib/ldap/XWikiLDAPConfig.java

```diff
@@ -626,6 +626,7 @@ public int getLDAPPort(XWikiContext context)
                 } else {
                     if (c == '\\') {
                         escaped = true;
+                        mapping.append(c);
```

Comment:
This change does not make sense.

It seems you missed that `\` is an escaping character, that means that if you need one in the end you need to escape it (so you need two) exactly like you would do in a Java String.

Reply:
Looking at the other changes it seems that `\` has been made a separator as well, so it is not longer used as an escape character. So the above would work if one also sets `escaped = false`. However in that case one has no longer the possibility to escape any characters.

I looked into the issue, too, and I think we have two escaping issues here, both not related to the actual LDAP escaping when using the values as queries:

- using the backslash as escape character when storing the values in the java code
- using the backslash as escape character when showing the values in the front end

In my opinion the problem is not here in the backend, but that the escape character is ignored when showing the values in the frontend. There the values are just split at `|`, ignoring all escaping.

That is why I changed the JavaScript code in the frontend to handle this in commit https://github.com/xwiki-contrib/ldap/commit/d525963ac3c2c8037bc5e167c90534333deca8d5
This has not been done for the AD Authenticator reusing the LDAP java code, but not the JavaScript part. I think the issue should be fixed in the JavaScript code of the AD Admin UI, not here.

Reply:
Helo, @tmortagne. The problem I have with this function, with regards to Active Directory application, manifests in the following way (on OpenLDAP):

* I create a group with the following DN: `cn=\#New Test,ou=Groups,dc=practice,dc=net` that gets automatically converted to `cn=\23New Test,ou=Groups,dc=practice,dc=net` (or to `cn=\\\#New Test,ou=Groups,dc=practice,dc=net` on windows AD)
* I associate a XWiki group to the DN as `XWiki.NewGroup=cn=\23New Test,ou=Groups,dc=practice,dc=net` in the XWikiPreferences document

Now, when I call the `getGroupMappings` function, it returns a map as such: 
`XWiki.NewGroup` -> `'cn=23New Test,ou=Groups,dc=practice,dc=net` (or `'cn=\#New Test,ou=Groups,dc=practice,dc=net` in AC) which is incorrect. 

The change I made fixed my issue.

Could you walk me over why [this line](https://github.com/xwiki-contrib/ldap/blob/master/ldap-authenticator/src/main/java/org/xwiki/contrib/ldap/XWikiLDAPConfig.java#L627) is needed? So I can better understand.